internal/pkg/runtime/engine/singularity/prepare_linux.go

```diff
@@ -519,6 +519,18 @@ func (e *EngineOperations) prepareContainerConfig(starterConfig *starter.Config)
 	// always set mount namespace
 	e.EngineConfig.OciConfig.AddOrReplaceLinuxNamespace(specs.MountNamespace, "")
 
+	// if IPC namespace is not allowed remove it from namespaces
+	if !e.EngineConfig.File.AllowIpcNs && e.EngineConfig.OciConfig.Linux != nil {
+		namespaces := e.EngineConfig.OciConfig.Linux.Namespaces
+		for i, ns := range namespaces {
```

Comment:
This loop, to remove a namespace, is now repeated 3 times. It would be good, at this point, to have a function that implements the removal of a specified namespace that can be called instead of repeating the code.